hoomd/md/EvaluatorPairTable.h

```diff
@@ -0,0 +1,232 @@
+    Both the potentials and forces** are provided the tables V(r) and F(r) at discreet \a r values between \a rmin and
+    \a rcut. Evaluations are performed by simple linear interpolation, thus why F(r) must be explicitly specified to
+    avoid large errors resulting from the numerical derivative. Note that F(r) should store - dV/dr.
+
+    \b Table memory layout
+
+    V(r) and F(r) are specified for each unique particle type pair. Three parameters need to be stored for each
+    potential: rmin, rcut, and dr, the minimum r, maximum r, and spacing between r values in the table respectively.
+    V(0) is the value of V at r=rmin. V(i) is the value of V at r=rmin + dr*i where i is chosen
+    such that r >= rmin and r < rcut. V(r) for r < rmin and >= rcut is 0. The same goes for F. Thus V and F are defined
+    between the region [rmin, rcut).
+
+    \b Interpolation
+    Values are interpolated linearly between two points straddling the given r. For a given r, the first point needed, i
+    can be calculated via i = floorf((r - rmin) / dr). The fraction between ri and ri+1 can be calculated via
+    f = (r - rmin) / dr - Scalar(i). And the linear interpolation can then be performed via V(r) ~= Vi + f * (Vi+1 - Vi)
+    \ingroup computes
+*/
+class EvaluatorPairTable
+    {
+    public:
+        //! Define the parameter type used by this pair potential evaluator
+        struct param_type
+            {
+            size_t width;   //!< the distance between table indices
+            Scalar rmin;    //!< the distance of the first index of the table potential
+            ManagedArray<Scalar> V_table; //!< the tabulated energy
+            ManagedArray<Scalar> F_table; //!< the tabulated force specifically - (dV / dr)
+
+            DEVICE void load_shared(char*& ptr, unsigned int& available_bytes) const { }
+            #ifdef ENABLE_HIP
+            //! Set CUDA memory hints
+            void set_memory_hint() const
+                {
+                // default implementation does nothing
+                }
+            #endif
+
+            #ifndef __HIPCC__
+            param_type() : width(0), rmin(0.0), V_table({}), F_table({}) {}
+
+            param_type(pybind11::dict v, bool managed=false)
+                {
+                const auto V_py = v["V"].cast<pybind11::array_t<Scalar>>().unchecked<1>();
+                const auto F_py = v["F"].cast<pybind11::array_t<Scalar>>().unchecked<1>();
+                if (V_py.size() != F_py.size())
+                    {
+                    throw std::runtime_error("The length of V and F arrays must be equal");
+                    }
+                width = V_py.size();
+                rmin = v["r_min"].cast<Scalar>();
+                V_table = ManagedArray<Scalar>((unsigned int)width, managed, 32);
+                F_table = ManagedArray<Scalar>((unsigned int)width, managed, 32);
```

Comment:
C++ style casts are more descriptive of the expected type of cast and can catch more programming errors. It is also more clear for come complex expressions (which this is not admittedly).
```suggestion
                V_table = ManagedArray<Scalar>(static_cast<unsigned int>(width), managed, 32);
                F_table = ManagedArray<Scalar>(static_cast<unsigned int>(width), managed, 32);
```